boxoffice/models/user.py

```diff
@@ -24,14 +27,52 @@ def default_user(context):
     return g.user.id if g.user else None
 
 
+def naive_to_utc(dt, timezone=None):
+    """
+    Returns a UTC datetime for a given naive datetime or date object
+    Localizes it to the given timezone and converts it into a UTC datetime
+    """
+    if timezone:
+        if isinstance(timezone, six.string_types):
+            tz = pytz.timezone(timezone)
+        else:
+            tz = timezone
+    elif isinstance(dt, datetime) and dt.tzinfo:
+        tz = dt.tzinfo
+    else:
+        tz = pytz.UTC
+
+    return tz.localize(dt).astimezone(tz).astimezone(pytz.UTC)
+
+
 class Organization(ProfileBase, db.Model):
     __tablename__ = 'organization'
-    __table_args__ = (db.UniqueConstraint('contact_email'),)
+    __table_args__ = (db.UniqueConstraint('contact_email'),
+        db.CheckConstraint('fy_start_month >= 1 and fy_start_month <= 12', 'org_month_check'),
+        db.CheckConstraint('fy_start_day >= 1 and fy_start_month <= 31', 'org_day_check'))
 
-    # The currently used fields in details are address(html), cin (Corporate Identity Number), pan, service_tax_no, support_email,
+    # The currently used fields in details are address(html), cin (Corporate Identity Number)
+    # pan service_tax_no, support_email,
     # logo (image url), refund_policy (html), ticket_faq (html), website (url)
     details = db.Column(JsonDict, nullable=False, server_default='{}')
     contact_email = db.Column(db.Unicode(254), nullable=False)
+    fy_start_month = db.Column(db.Integer, nullable=True)
+    fy_start_day = db.Column(db.Integer, nullable=True)
```

Comment:
If it's going to be changed to Jan 1, I suppose this will be announced _before_ Jan 1? In which case, we'll just need to update it on the 31st night, and `fy_start_at` will insure that invoice numbers will start over from 1.

Also, since this is an exception case, we'll need to manually intervene. But, apart from this case, `fy_start_at` will ensure the correct datetime every year, once `fy_start_month` and `fy_start_day` are set.